python/tests/pyspark/lib/spark_session.py

```diff
@@ -0,0 +1,39 @@
+import os
+from pyspark.sql import SparkSession
+
+
+def spark_session():
+    """
+    Minimize parallelism etc. to speed up execution.
+    Test data is small & local, so just avoid all kind of overhead.
+    """
+    builder = SparkSession.builder \
+        .config('spark.sql.shuffle.partitions', 1) \
+        .config('spark.default.parallelism', 1) \
+        .config('spark.shuffle.compress', False) \
+        .config('spark.rdd.compress', False)
+
+    # spark.jars doesn't seem to support classes folders. hence using the two extraClassPath properties.
+    # difference is that spark.jars takes a comma-separated list of jars,
+    #   while extraClassPath requires file paths with a platform-specific separator
+    classpath = _mleap_classpath()
+    builder = builder \
+        .config('spark.driver.extraClassPath', classpath) \
+        .config('spark.executor.extraClassPath', classpath)
+
+    return builder \
+        .enableHiveSupport() \
+        .getOrCreate()
+
+
+def _mleap_classpath():
+    """
+    Read classpath for mleap-spark-extension with the locally compiled classes.
+
+    Classpath file can be refreshed by running `sbt mleap-spark-extension/writeRuntimeClasspathToFile`.
+    However, that's only needed if making changes to dependencies.
+    """
+    classpath_file = os.path.join(
+        os.path.dirname(__file__), '..', '..', '..', '..', 'mleap-spark-extension', 'target', 'classpath-runtime.txt')
```

Comment:
Building a path this way looks brittle, what happens if someone runs the tests from a different folder? 
I'd aim for an absolute classpath. Also It'd be great if you could pass this classpath as an environment variable rather than a file somewhere on disk

Reply:
> what happens if someone runs the tests from a different folder?

It always works. Because `__file__` is used to refer to `classpath-runtime.txt` with a _relative_ path.

Reply:
> an environment variable rather than a file somewhere on disk

I think caching in a file is a must. For example, when I'm developing a pyspark test I don't want the additional overhead of invoking sbt every time that I run the test again.

For me it takes ~10s to run either of:
- `sbt mleap-spark-extension/writeRuntimeClasspathToFile`
- `sbt mleap-spark-extension/runtime:fullClasspath`

Reply:
You don't have to invoke sbt every time you run the tests, you could run it once and copy paste the output to an env. variable or your personal file on disk as it fits best your development flow. This code here makes a very strong move towards a way to develop and run tests that goes through a file-cache. Not everyone might use a similar environment, IDE or development-process, that's why I am suggesting to leave this less opinionated and flexible (for example I don't use an IDE). I'd rather let devs run sbt once, then they can paste it wherever it fits best for them

Reply:
I'm not fully sure what kind of flow this would be:

> copy paste the output to an env. variable or your personal file on disk

My aim was to make the tests work out of the without manual tweaking or configuring. Do you have some idea how that could be achieved otherwise?

Pyspark tests will any way depend on some generated files (classes), so is it any worse to write a file that has the required classpath?